ogrgrass.h

```diff
@@ -51,41 +50,45 @@ class OGRGRASSLayer final : public OGRLayer
     virtual ~OGRGRASSLayer();
 
     // Layer info
-    OGRFeatureDefn *GetLayerDefn() override
+    auto GetName() -> const char * override
+    {
+        return osName.c_str();
+    }
+    auto GetLayerDefn() -> OGRFeatureDefn * override
     {
         return poFeatureDefn;
     }
-    GIntBig GetFeatureCount(int) override;
-    OGRErr GetExtent(OGREnvelope *psExtent, int bForce) override;
-    virtual OGRErr GetExtent(int iGeomField, OGREnvelope *psExtent,
-                             int bForce) override
+    auto GetFeatureCount(int) -> GIntBig override;
+    auto GetExtent(OGREnvelope *psExtent, int bForce) -> OGRErr override;
+    virtual auto GetExtent(int iGeomField, OGREnvelope *psExtent, int bForce)
+        -> OGRErr override
     {
         return OGRLayer::GetExtent(iGeomField, psExtent, bForce);
     }
-    virtual OGRSpatialReference *GetSpatialRef() override;
-    int TestCapability(const char *) override;
+    virtual auto GetSpatialRef() -> OGRSpatialReference * override;
+    auto TestCapability(const char *) -> int override;
 
     // Reading
     void ResetReading() override;
-    virtual OGRErr SetNextByIndex(GIntBig nIndex) override;
-    OGRFeature *GetNextFeature() override;
-    OGRFeature *GetFeature(GIntBig nFeatureId) override;
+    virtual auto SetNextByIndex(GIntBig nIndex) -> OGRErr override;
+    auto GetNextFeature() -> OGRFeature * override;
```

Comment:
It fits with the type annotations in Python which is nice (although `auto` being in place of `def` is little strange). I just haven't seen a guideline suggesting to apply it generally. With a automated check available - at least in clang-tidy - I can see this can be realistically applied to the code base.